stack_collapse/src/stack_collapse.rs

```diff
@@ -5,38 +5,33 @@ use measureme::{Event, TimestampKind};
 
 pub fn collapse_stacks<'a>(
     events: impl Iterator<Item = Event<'a>>,
-    first_event_time: SystemTime,
```

Comment:
Ah, so the bug is basically that we should have been tracking the "first event time" of each thread individually instead of looking at the first event across all threads? The effect of that would make sense given the reports I've seen; the longer the crate takes to get to codegen (at which point, multiple threads come into play), the more unaccounted for time that was being reported.

Reply:
yes but also track the next_observation_time need to be tracked per thread as there can be out of order events between threads